### ISSUE TITLE
How to format information in Readme or wiki

### ISSUE THREAD

**Issue by pannufle** [akb89/nima#49]
Let's discuss here on the format the readme and wiki should have to be consistent across the different projects.
Here are a few example of questions we should answer:
- Do we have enough information to use the wiki or do we only keep the readme ?
- What sections the readme/wiki should have ?
- What information should go where ?
- ...

Add your questions and answers in the comments !

**Comment by candidj0** [comment]
As we discussed last time, we should keep the readme for instructions on how to start the project and the wiki to explain the project, its architecture, the research done and maybe what was done/what remains to be done? 

We also need to comment the project in order to generate a doc via ... (don't remember the name)?

**Comment by mariro8** [comment]
The readme should also explain what the project is about and its architecture. The wiki is for detailed informations like the research done in a specific context, why we decided to use this and not that etc.

**Comment by akb89** [comment]
Check out [Sphinx](http://www.sphinx-doc.org/en/master/) to automatically generate documentation from docstrings in Python. A more exhaustive list of tools is available [here](https://wiki.python.org/moin/DocumentationTools)

**Comment by loickouao** [comment]
Do we keep in the architecture readme or not?

for the readme I propose to add the description of the project and the information on how to run the code.
for the wiki the information on the architecture and following the plan of the overview and the strategies (What has been developed and what remains to be developed)
and not forgetting the information on good practice on how to code among other things the fact of lint and test your code before commit

**Comment by beny3** [comment]
I agree, it sounds good. Except complete architecture, because it might change, or not be validated. Plus, we'll generate it with sphinx or [doxigen](http://www.stack.nl/~dimitri/doxygen/).
Can 

- Research done

- Complete overview done by team1

- Strategy / Roadmap

be sections names in the wiki?

**Comment by akb89** [comment]
Note: 
- architecture is not the same thing as a the API documentation generated by Sphinx. 
- the overview is redundant with the wiki
- same for the strategy if you have properly documented language and application specs + todos

**Comment by akb89** [comment]
@pannufle 
for the README, I would suggest:
1. Brief description
2. HowTo (run, etc.)
3. todos (a shortlist of functionalities implemented and to be implemented)

for the wiki I would suggest:
1. Longer description (in intro)
2. Languages descriptions and specificities (for the task(s) at hand).
3. Overall architecture details
4. Overall specs (functionalities, toolkit choices, etc.)
5. Todos 
6. Building / Linting / Testing for code quality
What do you think? What would you add/remove?

**Comment by candidj0** [comment]
I agree with @pannufle Pannufle, a possible plan could be: 
ReadMe:
 - Specification
 - HowTo
 - Progress

Wiki:
- Purposes / what the project is supposed to do
- Architecture (schema + explanations)
- Detailed roadmap 
- Research 
          - research topic, summary, what it brings to the project (advantages/disadvantages) and how to include it in the existing project.

**Comment by akb89** [comment]
@candidj0 don't put the specs in the README, rather a brief intro. 
Also, research should preferably come first to explain choices introduced later on

**Comment by candidj0** [comment]
@akb89 yes it was the idea for spec.
For Wiki I just proposed main sections, I didn't really think about the order but you're right.

**Comment by akb89** [comment]
Goal of each documentation:
1. README: The readme should briefly introduce the project to an outsider 
1.a. brief description: what is the app, what it does;
1.b its current testing/development state, e.g. is the project still maintained and well tested?; 
1.c how to install/build/run the aplpication
1.d examples of use cases

**Comment by akb89** [comment]
Final recommendations (order of sections IS important)
README, I would suggest:
1. Brief description
2. HowTo (run, etc.)
3. todos (a shortlist of functionalities implemented and to be implemented)

for the wiki I would suggest:
1. Longer description (in intro)
2. Languages descriptions and specificities (for the task(s) at hand).
3. Overall architecture details
4. Overall specs (functionalities, toolkit choices, etc.)
5. Todos 
6. Building / Linting / Testing for code quality
